client/src/partials/accounts/accounts.js

```diff
@@ -1,193 +1,148 @@
-    $scope.model = model;
-    sortAccountList(model.account.data);
+  // fired on startup
+  function startup() {
+    // start up loading indicator
+    vm.loading = true;
+
+    // load Account Type
+    accountTypeService.getAccountType().then(function (data) {
+      vm.accountTypes = data;
+    }).catch(handler);
+
+    // load Cost Center
+    costCenterService.read().then(function (data) {
+      vm.costCenters = data;
+    }).catch(handler);
+
+    // load Profit Center
+    profitCenterService.read().then(function (data) {
+      vm.profitCenters = data;
+    }).catch(handler);
+
+    // load Reference
+    referenceService.read().then(function (data) {
+      vm.references = data;
+    }).catch(handler);
+
+    // load accounts
+    refreshAccounts();
   }
 
-  function submitAccount(account) {
-    //kill if account exists for now
-    if ($scope.model.account.get(account.number)) {
-      return accountError.throw('ERR_ACCOUNT_EXISTS', account.number);
-    }
-
-    //format account
-    var classe = account.number.substr(0,1);
-
-    var formatAccount = {
-      account_type_id: account.type.id,
-      account_number: account.number,
-      account_txt: account.title,
-      is_asset: account.is_asset,
-      is_ohada: account.is_ohada,
-      is_charge : account.is_charge,
-      is_used_budget: account.is_used_budget,
-      cc_id   : account.cc_id,
-      pc_id   : account.pc_id,
-      enterprise_id: appstate.get('enterprise').id,
-      parent: account.parent,
-      classe: account.number.substr(0,1),
-      reference_id : account.reference_id,
-      is_brut_link : account.is_brut_link
-    };
-
-    connect.post('account', [formatAccount])
-    .then(refreshAccountList)
-    .then(function () {
-      messenger.success($translate.instant('CONFIG_ACCOUNTING.SAVE_SUCCES'));
-      $scope.newAccount = {};
-      session.state = 'display';
-    });
+  function cancel() {
+    vm.view = 'default';
   }
-
-  function updateState(newState) {
-    session.state = newState;
+  
+  //This function cancels the information that should not exist 
+  //in the event of one or another type of accounts (balance or income/expense)  
+  function dataByTypes(){
+    if(vm.account.type.type === 'balance'){
+      vm.account.is_charge = null;
+      vm.account.cc_id = null;
+      vm.account.pc_id = null;
+    } else if (vm.account.type.type === 'income/expense'){
+      vm.account.is_asset = null;
+    }
   }
 
-  function formatCenter (c) {
-    return '' + c.text;
+  //This function first looks up the name type of account with the ID 
+  //and then cancels the information that should not exist in the event of one 
+  //or another type of accounts (balance or operating account)
+  function getTypeText(typeId, accountTypes){
+    var accountType = accountTypes.filter(function (item) {
+      return item.id  === typeId;
+    });  
+    vm.account.type = accountType[0].type;
+
+    if(vm.account.type === 'balance'){
```

Comment:
J'utilise else if pour prévenir l'alternative où vm.account.type est différent de balance and différent de income/expense